File: Quest/Assets/Src/Cards/Event Cards/ProsperityEvent.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ProsperityEvent : EventCard
{

    public ProsperityEvent() : base("Prosperity Throughout the Kingdom") { }

    public override void doEffect(GameState gState)
    {
        //Each player draws 2 cards
        foreach (Player player in gState.getPlayers())
        {
            for (int i = 0; i < 2; i++)
                player.AddCardToHand((AdventureCard)gState.getAdventureDeck().Draw());
        }
    }
}

File: eclipse/Quest/src/model/cards/eventCards/ProsperityEvent.java
package model.cards.eventCards;

import model.cards.AdventureCard;
import model.cards.EventCard;
import model.game.GameState;
import model.player.Player;

public class ProsperityEvent extends EventCard {

	public ProsperityEvent() {
		super("Prosperity Throughout the Realm");
	}

    public void doEffect(GameState gState)
    {
        //Each player draws 2 cards
        for (Player player : gState.getPlayers())
        {
            for (int i = 0; i < 2; i++)
                player.AddCardToHand((AdventureCard)gState.getAdventureDeck().Draw());
        }
    }
    
}
